### ISSUE TITLE
airflow(2.0.2): kubernetes_executor task never recived a scheduler_job_id

### ISSUE THREAD

**Issue by Z-Ghostshell** [apache/airflow#15883]
**Apache Airflow version**:

```
root@airflow-865c58dc6-cl754:~/airflow# airflow version
2.0.2
```

**Kubernetes version (if you are using kubernetes)** (use `kubectl version`):

```
kubectl version
...
Server Version: version.Info{Major:"1", Minor:"18+", GitVersion:"v1.18.9-eks-d1db3c", GitCommit:"d1db3c46e55f95d6a7d3e5578689371318f95ff9", GitTreeState:"clean", BuildDate:"2020-10-20T22:18:07Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}
```

**Environment**:

- **Cloud provider or hardware configuration**: eks
- **OS** (e.g. from /etc/os-release): 
```
PRETTY_NAME="Debian GNU/Linux 10 (buster)"
NAME="Debian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
```
- **Kernel** (e.g. `uname -a`):
- **Install tools**:
- **Others**:

**What happened**:
After upgrade to airflow 2.0.2 kubernetes_executor always report `airflow.exceptions.AirflowException: Could not get scheduler_job_id`

```
root@airflow-865c58dc6-cl754:~/airflow# airflow tasks run demo_bash_op bash_op 2021-05-16
/usr/local/lib/python3.8/site-packages/airflow/configuration.py:332 DeprecationWarning: The logging_level option in [core] has been moved to the logging_level option in [logging] - the old setting has been used, but please update your config.
/usr/local/lib/python3.8/site-packages/airflow/configuration.py:332 DeprecationWarning: The base_log_folder option in [core] has been moved to the base_log_folder option in [logging] - the old setting has been used, but please update your config.
/usr/local/lib/python3.8/site-packages/airflow/configuration.py:332 DeprecationWarning: The statsd_on option in [scheduler] has been moved to the statsd_on option in [metrics] - the old setting has been used, but please update your config.
[2021-05-16 18:15:53,229] {dagbag.py:451} INFO - Filling up the DagBag from /root/airflow/dags
Running <TaskInstance: demo_bash_op.bash_op 2021-05-16T00:00:00+00:00 [None]> on host airflow-865c58dc6-cl754
Traceback (most recent call last):
  File "/usr/local/bin/airflow", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/site-packages/airflow/__main__.py", line 40, in main
    args.func(args)
  File "/usr/local/lib/python3.8/site-packages/airflow/cli/cli_parser.py", line 48, in command
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/airflow/utils/cli.py", line 89, in wrapper
    return f(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/airflow/cli/commands/task_command.py", line 235, in task_run
    _run_task_by_selected_method(args, dag, ti)
  File "/usr/local/lib/python3.8/site-packages/airflow/cli/commands/task_command.py", line 68, in _run_task_by_selected_method
    _run_task_by_executor(args, dag, ti)
  File "/usr/local/lib/python3.8/site-packages/airflow/cli/commands/task_command.py", line 91, in _run_task_by_executor
    executor.start()
  File "/usr/local/lib/python3.8/site-packages/airflow/executors/kubernetes_executor.py", line 473, in start
    raise AirflowException("Could not get scheduler_job_id")
airflow.exceptions.AirflowException: Could not get scheduler_job_id
```

**What you expected to happen**:

**How to reproduce it**:

Just run `airflow task run` a task

**Anything else we need to know**:

**Comment by fengyu05** [comment]
seems to be relative to https://github.com/apache/airflow/issues/13805 which suppose to be fixed in the 2.0.2

**Comment by Reddy1990** [comment]
HI All, I'm still getting the same error. While using CeleryKubernetesExecutor

____ |( )_______ / /________ __
____ /| |_ /__ / / __ / __ _ | /| / /
___ ___ | / _ / _ / _ / / // / |/ |/ /
// |// // // // _/____/|__/
[2021-07-09 10:31:52,972] {scheduler_job.py:1251} INFO - Starting the scheduler
[2021-07-09 10:31:52,972] {scheduler_job.py:1256} INFO - Processing each file at most -1 times
[2021-07-09 10:31:52,978] {kubernetes_executor.py:471} INFO - Start Kubernetes executor
[2021-07-09 10:31:52,978] {scheduler_job.py:1302} ERROR - Exception when executing SchedulerJob._run_scheduler_loop
Traceback (most recent call last):
File "/home/airflow/.local/lib/python3.6/site-packages/airflow/jobs/scheduler_job.py", line 1276, in _execute
self.executor.start()
File "/home/airflow/.local/lib/python3.6/site-packages/airflow/executors/celery_kubernetes_executor.py", line 60, in start
self.kubernetes_executor.start()
File "/home/airflow/.local/lib/python3.6/site-packages/airflow/executors/kubernetes_executor.py", line 473, in start
raise AirflowException("Could not get scheduler_job_id")
airflow.exceptions.AirflowException: Could not get scheduler_job_id
[2021-07-09 10:31:52,979] {dag_processing.py:453} WARNING - Ending without manager process.
[2021-07-09 10:31:52,979] {scheduler_job.py:1305} INFO - Exited execute loop

Environment:

Using Airflow 2.0.2 Version
Kubernetes v1.20.7
Cloud provider or hardware configuration: AWS Linux

I was using KubernetesExecutor (all was working fine , faced no issues and able to run DAG's/Jobs) and now experimenting with CeleryKubernetesExecutor the issue started showing up and I check if any job/Dag/task is causing the issue where the Scheduler looks for but I tried after deleting DAG's and placing any empty dir but the issue is still there.

**Comment by potiuk** [comment]
I believe there were a few fixes to CeleryKubernetesExecutor including some final fixes in 2.1.2 (which will be released soon). Could you plase double-check @Reddy1990 and @fengyu05 if the issue will be fixed there after we release the Release Candidate in the next few days?
